enterprise/backend/src/metabase_enterprise/scim/v2/api.clj

```diff
@@ -188,98 +206,103 @@
   {start-index  [:maybe ms/PositiveInt]
    c            [:maybe ms/PositiveInt]
    filter-param [:maybe ms/NonBlankString]}
-  (let [limit          (or c default-pagination-limit)
-        ;; SCIM start-index is 1-indexed, so we need to decrement it here
-        offset         (if start-index (dec start-index) default-pagination-offset)
-        filter-param   (when filter-param (codec/url-decode filter-param))
-        where-clause   [:and [:= :type "personal"]
-                        (when filter-param (user-filter-clause filter-param))]
-        users          (t2/select (cons :model/User user-cols)
-                                  {:where    where-clause
-                                   :limit    limit
-                                   :offset   offset
-                                   :order-by [[:id :asc]]})
-        hydrated-users (t2/hydrate users :scim_user_group_memberships)
-        results-count  (count hydrated-users)
-        items-per-page (if (< results-count limit) results-count limit)
-        result         {:schemas      [list-schema-uri]
-                        :totalResults (t2/count :model/User {:where where-clause})
-                        :startIndex   (inc offset)
-                        :itemsPerPage items-per-page
-                        :Resources    (map mb-user->scim hydrated-users)}]
-    (scim-response result)))
+  (with-prometheus-counters
+    (let [limit          (or c default-pagination-limit)
+          ;; SCIM start-index is 1-indexed, so we need to decrement it here
+          offset         (if start-index (dec start-index) default-pagination-offset)
```

Comment:
Will limit be off-by-one if start-index gets dec'd?

Reply:
I don't think so? limit just controls the number of values fetched at once and will be 100 (default value) for most calls. So the client will send `start-index=1`, `start-index=101`, etc, and we'll fetch the same users using offsets of 0, 100, etc.